local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -532,14 +518,10 @@ void LocalPlannerNode::printPointInfo(double x, double y, double z) {
 
 void LocalPlannerNode::pointCloudCallback(
     const sensor_msgs::PointCloud2::ConstPtr& msg, int index) {
+  std::lock_guard<std::mutex> lck(*(cameras_[index].cloud_msg_mutex_));
   cameras_[index].newest_cloud_msg_ = *msg;  // FIXME: avoid a copy
   cameras_[index].received_ = true;
-
-  {
-    std::unique_lock<std::mutex> lck(*(cameras_[index].cloud_ready_mutex_));
-    cameras_[index].transformed_ = false;
```

Comment:
I think it should work never the less: You are right, if a new cloud is comming in before the last transformed one is copied into the planner `transformed_` is still true. This means that the cloud can still be copied into the planner (which is ok, because `cameras_[index].pcl_cloud` still contains the last transformed cloud) the transforming of the new cloud will be blocked until the last one is copied into the planner and `transformed_` is set to false.

Do you agree?

Reply:
Yes, so if the new frame comes, it still use last transformed frame.